src/components/workspace/ControlBar.tsx

```diff
@@ -196,7 +197,11 @@ class ControlBar extends React.PureComponent<ControlBarProps, { value: string }>
           this.props.handleSetEditorSessionId!('');
         }
       };
-      xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/' + this.state!.value, true);
+      xmlhttp.open(
+        'GET',
+        'https://' + LINKS.SHAREDB_SERVER + 'gists/' + this.state!.joinElemValue,
```

Comment:
this should be externalized into another component leh we shouldn't be having logic in a dumb component such as the control bar.

Reply:
The control bar turns out to be interesting. I think it should have the requested logic, in order to support run/autorun. I feel that run/stop/autorun can be succinctly supported in the requested feature. Let's merge this and see if we get feedback from users.

Reply:
By externalized do you mean putting it into a helper file? If so we can do it in another PR.

Reply:
yes I meant another file! if you think about it if the url changes lets say `gists` or `lastest` get changed to something else the control bar is the definitely wrong place to come to change things. As prof always says think in levels of abstraction!

Reply:
I'll approve but at the rate we're going something is gonna break soon haha